### PR TITLE
Dockerize local development

### PR DESCRIPTION
**Pull request by 10thfloor**
- Local dev via Docker
- Production build via Docker
- Golang tests
- JS tests
- Cypress tests
- Github Action / Cypress tests for Docker

### CODE REVIEW

File: .github/workflows/integration-local.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+      - name: Run Cypress Tests
+        uses: cypress-io/github-action@v2
+        with:
+          install: false
+          start: npm run test
+          browser: chrome
+          headless: true
+          record: false
+          config: pageLoadTimeout=100000,baseUrl=http://localhost:3001,chromeWebSecurity=false
```
Comment (10thfloor) [inline .github/workflows/integration-local.yml:44]:
```suggestion
          config: baseUrl=http://localhost:3001,chromeWebSecurity=false
```

Comment (10thfloor) [reply .github/workflows/integration-local.yml:44]: The action itself has a 10 minute timeout so this is redundant.

File: .github/workflows/integration-local.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+name: Kitty Items Integrations Tests
```
Comment (psiemens) [inline .github/workflows/integration-local.yml:1]:
```suggestion
name: Kitty Items Integration Tests
```

File: .github/workflows/integration-local.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+      - uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: cypress-videos
+          path: cypress/videos
+
+      - name: Stop Containers
+        if: always()
+        run: docker-compose -f "docker-compose.yml" down
```
Comment (psiemens) [inline .github/workflows/integration-local.yml:58]:
```suggestion
        run: docker-compose down
```

File: README.md [hunk 4]
```diff
@@ -30,8 +28,25 @@ _⚠️ This project requires `flow-cli v0.15.0` or above._
...
+- Run `npm install` in the root of the project.
+- Run `npx lerna exec npm install` to install project dependencies.
+
+### 4. Start the project
+
+## Local development
+
+
+- Run `./srcipts/run-local.sh` 
+  - This script exports the necessary configuration and starts the `web` and `api` applications in local Docker containers. (See`docker-compose.yml` for details.)
```
Comment (psiemens) [inline README.md:43]:
```suggestion
  - This script exports the necessary configuration and starts the `web` and `api` applications in local Docker containers (see [docker-compose.yml](docker-compose.yml) for details).
```

File: README.md [hunk 5]
```diff
@@ -30,8 +28,25 @@ _⚠️ This project requires `flow-cli v0.15.0` or above._
...
+  - Local development uses the [Flow Emulator]() and the [FCL Dev-Wallet]() to simulate the blockchain and FCL compatible wallets.
+  - All contracts are deployed to the Flow emulator.
```
Comment (psiemens) [inline README.md:45]:
```suggestion
  - All contracts are deployed to the emulator.
```

File: README.md [hunk 6]
```diff
@@ -30,8 +28,25 @@ _⚠️ This project requires `flow-cli v0.15.0` or above._
...
+- Run `npx lerna exec npm install` to install project dependencies.
+
+### 4. Start the project
+
+## Local development
+
+
+- Run `./srcipts/run-local.sh` 
+  - This script exports the necessary configuration and starts the `web` and `api` applications in local Docker containers. (See`docker-compose.yml` for details.)
+  - Local development uses the [Flow Emulator]() and the [FCL Dev-Wallet]() to simulate the blockchain and FCL compatible wallets.
```
Comment (psiemens) [inline README.md:44]:
```suggestion
  - Local development uses the [Flow Emulator](https://docs.onflow.org/emulator/) and the [FCL Development Wallet](https://github.com/onflow/fcl-dev-wallet) to simulate the blockchain and an FCL-compatible wallet.
```

File: README.md [hunk 7]
```diff
@@ -30,8 +28,25 @@ _⚠️ This project requires `flow-cli v0.15.0` or above._
...
+  - All contracts are deployed to the Flow emulator.
+
 
-### 3. Create a Flow Testnet account
+## Testnet development
+#### Create a Flow Testnet account
```
Comment (psiemens) [inline README.md:49]:
```suggestion
### Create a Flow Testnet account 
```

**Review by psiemens — COMMENTED** [review]
🎉 

I spent some time debugging this branch and tracked down the `empty block IDs` error: https://github.com/onflow/flow-js-sdk/pull/714

There's a temporary fix in this branch that skips the first block (height=0), which is okay since there aren't any relevant events in that block.

I also added a [Dockerfile to the fcl-dev-wallet repo](https://github.com/onflow/fcl-dev-wallet/blob/main/Dockerfile) and published it to the container registry, which allowed me to remove the `fcl-dev-wallet` submodule from this repository.

With the emulator and dev wallet Dockerfiles gone, I moved the API and web Dockerfiles inside of their respective directories.

File: api/src/index.ts [hunk 8]
```diff
@@ -61,14 +61,17 @@ async function run() {
   );
 
   // Make sure we're pointing to the correct Flow Access API.
-  fcl.config().put("accessNode.api", config.accessApi);
+  fcl
+    .config()
+    .put("accessNode.api", config.accessApi)
+    .put("decoder.Type", (val) => val.staticType);
```
Comment (10thfloor) [inline api/src/index.ts:67]:
```suggestion
    .put("decoder.Type", val => val.staticType);
```

**Pull request merged**